packages/amplify_datastore/ios/Classes/types/model/FlutterModelField.swift

```diff
@@ -20,20 +20,56 @@ import Amplify
 public struct FlutterModelField {
     
     public let name: String
-    public let type: ModelFieldType
+    public let type: FlutterModelFieldType
     public let isRequired: Bool
     public let isArray: Bool
-    public let attributes: [ModelFieldAttribute] = []
-    public let association: ModelAssociation? = nil
-    public let authRules: AuthRules = []
+    public let association: FlutterModelAssociation?
+    public let authRules: [FlutterAuthRule]?
     
-    init(serializedData: [String: Any]) {
-        self.name = serializedData["name"] as! String
-        self.type = ModelFieldType.from(type: FlutterModelField.deserializeFieldType(type: serializedData["type"] as! String))
-        self.isRequired = serializedData["isRequired"] as! Bool
-        self.isArray = serializedData["isArray"] as! Bool
+    init(serializedData: [String: Any]) throws {
+
+        guard let name = serializedData["name"] as? String
+        else {
+            throw ModelSchemaError.parse(className: "FlutterModelField", fieldName: "name", desiredType: "String")
+        }
+        self.name = name
+
+        if let inputTypeMap = serializedData["type"] as? [String: Any]{
```

Comment:
use guard here as well?

Reply:
Do you know how to use guard if there's an inner statement?

`if let inputTypeMap = serializedData["type"] as? [String: Any]{
            **self.type = try FlutterModelFieldType(serializedData: inputTypeMap)** <-- this causes compile issues if using guard instead of if
        }`

Reply:
It will be like this:

```swift
guard let inputTypeMap = serializedData["type"] as? [String: Any] else {
       throw ModelSchemaError.parse(
                 className: "FlutterModelField",
                 fieldName: "type",
                 desiredType: "[String: Any]")
       return
}
self.type = try FlutterModelFieldType(serializedData: inputTypeMap)
```

Reply:
nice thanks you.  I don't think we need the return (warning shows) so I didn't include that part.